rdtools/system_analysis.py

```diff
@@ -0,0 +1,587 @@
+            bool_filter = bool_filter & f
+        if 'tcell_filter' in self.filter_params.keys():
+            if cell_temp is None:
+                raise ValueError('Cell temperature must be available to use tcell_filter')
+            f = filtering.tcell_filter(cell_temp, **self.filter_params['tcell_filter'])
+            bool_filter = bool_filter & f
+        if 'clip_filter' in self.filter_params.keys():
+            if self.pv_power is None:
+                raise ValueError('PV power (not energy) is required for the clipping filter. Either omit the clipping filter,'
+                                 'provide PV power at instantiation, or explicitly assign system_analysis.pv_power.')
+            f = filtering.clip_filter(self.pv_power, **self.filter_params['clip_filter'])
+            bool_filter = bool_filter & f
+        if 'ad_hoc_filter' in self.filter_params.keys():
+            if self.filter_params['ad_hoc_filter'] is not None:
+                bool_filter = bool_filter & self.filter_params['ad_hoc_filter']
+        if case == 'clearsky':
+            if self.poa is None or self.clearsky_poa is None:
+                raise ValueError('Both poa and clearsky_poa must be available to do clearsky filtering with csi_filter')
+            f = filtering.csi_filter(self.poa, self.clearsky_poa, **self.filter_params['csi_filter'])
+            bool_filter = bool_filter & f
+
+        if case == 'sensor':
+            self.sensor_filter = bool_filter
+        elif case == 'clearsky':
+            self.clearsky_filter = bool_filter
+
+    def aggregate(self, normalized, insolation):
+        '''
+        Returns insolation-weighted normalized PV energy and the associated aggregated insolation
+
+        Parameters
+        ----------
+        normalized: Pandas Times Series (numeric), normalized pv energy time series
+        insolation: Pandas Time Series (numeric), Time series of insolation associated
+                    with each normalize_energy point
+
+        Returns
+        -------
+        (aggregated, aggregated_insolation)
+        Tuple of Pandas Time Series consisting of insolation-weighted aggregated normalized PV energy
+        and the associated aggregated insolation.
+        '''
+        aggregated = aggregation.aggregation_insol(normalized, insolation, self.aggregation_freq)
+        aggregated_insolation = insolation.resample(self.aggregation_freq).sum()
+
+        return aggregated, aggregated_insolation
+
+    def yoy_degradation(self, aggregated, **kwargs):
+        '''
+        Perform year-on-year degradation analysis on insolation-weighted aggregated energy yield.
+
+        Parameters
+        ----------
+        aggregated: Pandas Time Series of insolation-weighted aggregated normalized PV energy
+        kwargs: passed to degradation.degradation_year_on_year()
+
+        Returns
+        -------
+        dict or year-on-year results with keys:
```

Comment:
typo in `dict or year-on-year results with keys:`?

Reply:
Fixed, thanks